### PR TITLE
Add the initial version of the Samples Gallery

### PR DESCRIPTION
**Pull request by nums11**
Adds the initial version of the samples gallery with 3 samples
- Image Classification
- Batching
- Image Effects

### CODE REVIEW

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/SampleBasePage.xaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    <ScrollViewer>
+        <Grid>
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="0.7*"></ColumnDefinition>
+                <ColumnDefinition Width="0.3*"></ColumnDefinition>
+            </Grid.ColumnDefinitions>
+            <StackPanel Grid.Row="0" Grid.Column="0" Padding="40,40">
+                <TextBlock FontWeight="Bold" FontSize="25" Text="{x:Bind sampleMetadata.Title}" />
+                <TextBlock FontSize="16" Margin="0,15,0,0" Text="{x:Bind sampleMetadata.Description}" />
```
Comment (linnealovespie) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/SampleBasePage.xaml:28]: Add a TextWrapping="Wrap" attribute to this textblock to help with smaller window sizes.

File: .gitmodules [hunk 2]
```diff
@@ -3,4 +3,4 @@
 	url = https://github.com/lutzroeder/Netron.git
 [submodule "Samples/RustSqueezenet/winrt-rs"]
 	path = Samples/RustSqueezenet/winrt-rs
-	url = https://github.com/microsoft/winrt-rs.git
+	url = https://github.com/microsoft/winrt-rs.git
```
Comment (martinb35) [inline .gitmodules:6]: is there something different with your line ending settings? seems like this file (and at least one other) shouldn't have changes in it.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery (Package)/Package.appxmanifest [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="42365107-cbb5-412e-80ac-554e5225ab17"
+    Publisher="CN=sheil"
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery (Package)/Package.appxmanifest:11]: can we change the "sheil" publisher here?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery (Package)/Package.appxmanifest [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    Version="1.0.0.0" />
+
+  <Properties>
+    <DisplayName>WinMLSamplesGallery (Package)</DisplayName>
+    <PublisherDisplayName>sheil</PublisherDisplayName>
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery (Package)/Package.appxmanifest:16]: let's use a different publisher name

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            return openPicker.PickSingleFileAsync().GetAwaiter().GetResult();
+        }
+        public static BitmapDecoder CreateBitmapDecoderFromStorageFile(StorageFile file)
+        {
+            if (file == null)
+            {
+                return null;
+            }
+
+            var stream = file.OpenAsync(FileAccessMode.Read).GetAwaiter().GetResult();
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs:53]: can we follow up on this "GetAwaiter()" pattern vs the "await" keyword? I don't understand when to use one or the other.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+namespace WinMLSamplesGallery.Common
+{
+    [ComImport, Guid("3E68D4BD-7135-4D10-8018-9FB6D9F33FA1"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    public interface IInitializeWithWindow
+    {
+        void Initialize([In] IntPtr hwnd);
+    }
+
+    public static class File
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs:20]: "File" feels too generic for the name of this class.

Comment (martinb35) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs:20]: same with the name of this file, "File.cs". also, there's an image helper later in this file, so not sure if "File.cs" accurately describes what's in here.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/HighResolutionClock.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        public TensorFloat ToTensorFloat()
+        {
+            return TensorFloat.CreateFromBuffer(new long[] { num_bytes_ }, this);
+        }
+
+        public unsafe void Buffer(out byte* pByte)
+        {
+            pByte = (byte*)ptr_;
+        }
+#pragma warning restore CA1416 // Validate platform compatibility
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/HighResolutionClock.cs:64]: can you remove the commented out code?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿// To learn more about WinUI, the WinUI project structure,
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs:1]: do we want these comments?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿// To learn more about WinUI, the WinUI project structure,
+// and more about our project templates, see: http://aka.ms/winui-project-info.
+
+namespace WinMLSamplesGallery.Controls
+{
+    public class Prediction
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs:6]: why is the "Prediction" class inside the "Controls" namespace? I would assume controls is about the UI, but the prediction object is not limited to UI usage, right?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/SamplesGrid.xaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
+<Page
+    x:Class="WinMLSamplesGallery.SamplesGrid"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:WinMLSamplesGallery"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/SamplesGrid.xaml:6]: are you using the blend/2008 namespace? same for others here. let's remove the ones we're not using.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using System.Collections.ObjectModel;
+
+// To learn more about WinUI, the WinUI project structure,
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs:17]: let's remove these comments

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+// and more about our project templates, see: http://aka.ms/winui-project-info.
+
+namespace WinMLSamplesGallery.Controls
+{
+    public sealed class Metric
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs:22]: "Metric" is a very generic name, can we make it more specific?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/SampleHeader.xaml.cs [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+using Windows.Foundation.Collections;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+
+// To learn more about WinUI, the WinUI project structure,
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/SampleHeader.xaml.cs:16]: remove

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/InputData/ilsvrc2013.txt [hunk 14]
```diff
@@ -0,0 +1,200 @@
+0,accordion
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/InputData/ilsvrc2013.txt:1]: can you explain the name of this file?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+                <NavigationViewItem Icon="Library" Content="Audio" Tag="audio" />
+                <NavigationViewItemSeparator />
+                <NavigationViewItem Icon="Library" Content="Benchmark" Tag="benchmark" />
+            </NavigationView.MenuItems>
+            <Frame x:Name="contentFrame" Navigated="contentFrame_Navigated" BorderBrush="#dbdbdb" BorderThickness="1,1,0,0" Margin="0,40,0,0"
+                   CornerRadius="5,0,0,0" Background="White" FontFamily="Arial"/>
+        </NavigationView>
+    </Grid>
+    
+
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml:40]: can you remove the empty spaces?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+        public static Frame mainFrame;
+
+        public MainWindow()
+        {
+            this.InitializeComponent();
+            this.Title = "Windows AI Samples Gallery";
+            mainFrame = contentFrame;
+        }
+
+        private void nvSample_ItemInvoked(NavigationView sender, NavigationViewItemInvokedEventArgs args)
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:41]: what is the "nv" prefix of "nvSample_ItemInvoked"? can you just spell out the whole word?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+        {
+            if (args.IsSettingsInvoked == true)
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:43]: this if block doesn't do anything

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        private void NavView_Navigate(
+            string navItemTag,
+            Microsoft.UI.Xaml.Media.Animation.NavigationTransitionInfo transitionInfo)
+        {
+            Type _page = null;
+            if (navItemTag == "settings")
+            {
+                // _page = typeof(SettingsPage);
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:60]: why is this commented out?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+            
+            contentFrame.GoBack();
+            return true;
+        }
+
+        private void contentFrame_Navigated(object sender, NavigationEventArgs e)
+        {
+            nvSample.IsBackEnabled = contentFrame.CanGoBack;
+
+            //if (contentFrame.SourcePageType == typeof(SettingsPage))
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:103]: can you remove or un-comment this?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+            // NavView doesn't load any page by default, so load home page.
+            nvSample.SelectedItem = nvSample.MenuItems.ElementAt(0);
+            // If navigation occurs on SelectionChanged, then this isn't needed.
+            // Because we use ItemInvoked to navigate, we need to call Navigate
+            // here to load the home page.
+            NavView_Navigate("home", new Microsoft.UI.Xaml.Media.Animation.EntranceNavigationTransitionInfo());
+
+            // Listen to the window directly so the app responds
+            // to accelerator keys regardless of which element has focus.
+            //Microsoft.UI.Xaml.Window.Current.CoreWindow.Dispatcher.
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:138]: can you remove or un-comment this?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml [hunk 21]
```diff
@@ -0,0 +1,14 @@
+﻿<Page
+    x:Class="WinMLSamplesGallery.Audio"
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml:2]: should we remove this page for now?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml.cs [hunk 22]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml.cs:1]: remove this file?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Benchmark.xaml [hunk 23]
```diff
@@ -0,0 +1,14 @@
+﻿<Page
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Benchmark.xaml:1]: what is this page for?

**Review by martinb35 — CHANGES_REQUESTED** [review]

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery (Package)/Package.appxmanifest (hunk 4 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery (Package)/Package.appxmanifest:16]: Should the publisher name be Windows ML?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs (hunk 5 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs:53]: I changed the batching sample to use the "await" keyword. @smk2007 preferred to use GetAwaiter instead but we can discuss which one works better

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs (hunk 19 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:103]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs (hunk 20 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:138]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml (hunk 21 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml:2]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml.cs (hunk 22 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Audio.xaml.cs:1]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs [hunk 24]
```diff
@@ -0,0 +1,448 @@
...
+        }
+#pragma warning restore CA1416 // Validate platform compatibility
+
+        public ImageClassifier()
+        {
+            this.InitializeComponent();
+
+            CurrentImageDecoder = null;
+            CurrentModel = Classifier.NotSet;
+            AllModelsGrid.SelectRange(new Microsoft.UI.Xaml.Data.ItemIndexRange(0, 1));
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:101]: isn't "Microsoft.UI.Xaml" already in the list of used namespaces?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs [hunk 25]
```diff
@@ -0,0 +1,448 @@
...
+                    { Classifier.ZFNet512,          () => TensorizationModels.TopK(TopK) },
+                };
+            }
+
+            if (_preProcessorDictionary == null)
+            {
+                _preProcessorDictionary = new Dictionary<Classifier, Func<LearningModel>>{
+                    { Classifier.SqueezeNet,        null }, // No preprocessing required
+                    { Classifier.MobileNet,         () => TensorizationModels.Normalize0_1ThenZScore(224, 224, 4,
+                                                                                                new float[] { 0.485f, 0.456f, 0.406f },
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:169]: we should have comments explaining where these numbers come from. Also, maybe have links to the research papers where the values come from?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs [hunk 26]
```diff
@@ -0,0 +1,448 @@
...
+
+            InitializeWindowsMachineLearning();
+        }
+
+        private void InitializeWindowsMachineLearning()
+        {
+            _tensorizationSession =
+                CreateLearningModelSession(
+                    TensorizationModels.BasicTensorization(
+                        224, 224,
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:206]: please name these values (height, width, batch size, channels, etc.)

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs [hunk 27]
```diff
@@ -0,0 +1,448 @@
...
+                        1, 4, CurrentImageDecoder.PixelHeight, CurrentImageDecoder.PixelWidth,
+                        "nearest"),
+                        LearningModelDeviceKind.Cpu);
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:209]: shouldn't this come from the combo box value?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs [hunk 28]
```diff
@@ -0,0 +1,682 @@
...
+        Blur3x3,
+        Blur5x5,
+        Blur7x7,
+        ResizeCubicFill,
+        ResizeCubicFit,
+        ResizeLinearFill,
+        ResizeLinearFit,
+        ResizeNearestFill,
+        ResizeNearestFit,
+        PixelSwizzle_123_321,
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs:35]: does the alpha channel never participate in the swizzles?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs [hunk 29]
```diff
@@ -0,0 +1,682 @@
...
+        }
+
+        private void DeviceComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            ApplyEffects();
+        }
+    }
+}
+/*
+  // Will need this for 0 copy interop
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs:640]: can we remove the commented out code?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs [hunk 30]
```diff
@@ -0,0 +1,692 @@
...
+            COMPLEX128 = 15,    // complex with float64 real and imaginary components
+
+            // Non-IEEE floating-point format based on IEEE754 single-precision
+            // floating-point number truncated to 16 bits.
+            // This format has 1 sign bit, 8 exponent bits, and 7 mantissa bits.
+            BFLOAT16 = 16,
+        }
+
+#pragma warning disable CA1416 // Validate platform compatibility
+        public static LearningModel IdentityNCHW()
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:37]: seems like each of these methods could use a comment saying what they do

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs [hunk 31]
```diff
@@ -0,0 +1,692 @@
...
+            var sliceStarts = new long[] { 0, 0, -1, 0 };
+            var sliceEnds = new long[] { long.MaxValue, long.MaxValue, long.MinValue, long.MaxValue };
+            var axes = new long[] { 0, 1, 2, 3 };
+            var sliceSteps = new long[] { 1, 1, -1, 1 };
+            return Slice(sliceStarts, sliceEnds, axes, sliceSteps);
+        }
+
+        public static LearningModel RotateRight90()
+        {
+            return Rotate90(true /*rotateRight = false*/);
```
Comment (martinb35) [inline Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:330]: rotateRight = true?

**Review by martinb35 — CHANGES_REQUESTED** [review]

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Benchmark.xaml (hunk 23 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Pages/Benchmark.xaml:1]: Seemed like a new page @smk2007 was going to add. Deleting for now.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs (hunk 24 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:101]: ItemIndexRange comes from the Microsoft.UI.Xaml.Data namespace which is not currently being included. I'll add a using statement.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs (hunk 25 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:169]: @smk2007 may be able to answer this

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs (hunk 26 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:206]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs (hunk 27 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:209]: It's possible that @smk2007 always wanted tensorization to be done on CPU?

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs (hunk 28 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs:35]: @smk2007 can better answer this

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs (hunk 29 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs:640]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs (hunk 30 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:37]: @smk2007 can better answer this

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs (hunk 31 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:330]: @smk2007 can better answer this

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs (hunk 6 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs:20]: Changed to ImageHelper

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs (hunk 25 above)
Comment (smk2007) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:173]: They come from the preprocessing steps described in the model zoo:
https://github.com/onnx/models/tree/master/vision/classification/mobilenet

Each model has its own preprocessing steps and values based on the model's definition.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs (hunk 27 above)
Comment (smk2007) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageClassifier/ImageClassifier.xaml.cs:213]: It is slower on dml.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs (hunk 28 above)
Comment (smk2007) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/ImageEffects/ImageEffects.xaml.cs:35]: alpha channel is sliced out

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs (hunk 30 above)
Comment (smk2007) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:37]: Yes, a comment will help.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs (hunk 31 above)
Comment (smk2007) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:330]: Yes, comment needs to be updated.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs (hunk 30 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/TensorizationModels.cs:37]: @smk2007 said he would update these but we shouldn't block on these for now

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs (hunk 17 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:41]: Removing.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs (hunk 18 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:58]: Not sure. There may have been plans for a settings page but I'll remove.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs (hunk 16 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml.cs:39]: I think 'nv' is short for navigation. I think this was just the standard followed in the navigation demo for WinUI

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml (hunk 15 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/MainWindow.xaml:40]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/SampleHeader.xaml.cs (hunk 13 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/SampleHeader.xaml.cs:16]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs (hunk 12 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs:22]: Changing to PerformanceMetric

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs (hunk 11 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Controls/PerformanceMonitor.xaml.cs:17]: Done.

**Review by martinb35 — APPROVED** [review]

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs (hunk 9 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs:6]: It is only used for showing the UI in the ImageClassification sample

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs (hunk 8 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/Prediction.cs:1]: Removed.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/HighResolutionClock.cs (hunk 7 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/HighResolutionClock.cs:64]: Done.

File: Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/SamplesGrid.xaml (hunk 10 above)
Comment (nums11) [reply Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/SamplesGrid.xaml]: Done.

**Pull request merged**